### ISSUE TITLE
Question: flow control and conditionals

### ISSUE THREAD

**Issue by djfinnoy** [tektoncd/pipeline#3819]
I can't find much information about using basic code flow control within Tekton tasks; I'm guessing this isn't supported yet (or I'm just bad at searching the internet).
Below I will describe a use case, my question is whether what I want to do is possible right now, and if not; whether it's on the roadmap for future releases.

Consider [the kaniko task from the catalog](https://github.com/tektoncd/catalog/blob/master/task/kaniko/0.1/kaniko.yaml):
I'd like to push multiple tags to my container registry.
This means adding multiple `--destination` flags to the `build-and-push` step.

At the moment, it looks like I have to hardcode the number of destinations, and make a param for each one, eg:

```
...
  params:
    ...
    - name: IMAGE_TAG_1
      value: hello
    - name: IMAGE_TAG_2
      value: world
 ...
   steps:
     - name: build-and-push
       ...
       command:
          ...
          - --destination=$(params.IMAGE):$(params.IMAGE_TAG_1)
          - --destination=$(params.IMAGE):$(params.IMAGE_TAG_2)
```
This isn't nice. I'd prefer to specify an array of tags, and have some sort of loop under the step that adds a `--destination` for each tag.
Are there any better ways of achieving this than the solution outlined above?

**Comment by afrittoli** [comment]
Hi @djfinnoy, Tekton does support parameters of type array - but they are very limited - you cannot expand an array to a string or interpolate the values of the array with other strings, so I would not recommend this route.

The catalog task supports `EXTRA_ARGS` so you could pass a `--destination=$(params.IMAGE):$(params.IMAGE_TAG_1) --destination=$(params.IMAGE):$(params.IMAGE_TAG_2)` in there. 

You could have an altered version of the catalog task where you could pass something like a comma separated list of tasks.
The `kaniko` debug image includes a shell, so you could use that to allow parsing the list and building the kaniko command within the step script.

We have experimental support for task loops i.e. running a task multiple times with different inputs. You could build the image once and then push multiple tags with a loop of tasks, but the overhead of creating a pod just to tag an image might be too much.

In our release process we use `crane` to tag images and push them to the different geographies of the container registry:
https://github.com/tektoncd/pipeline/blob/1cd612f90db17dbfc83e37b99fd8ce6b6a07466b/tekton/publish.yaml#L160-L198

hth

**Comment by djfinnoy** [comment]
Hah, version 0.2 was released after I posted my question, the `EXTRA_ARGS` param was not included in version 0.1.
Thanks for pointing it out, your suggestion to include a second `--destination` via `EXTRA_ARGS` works perfectly.
